File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let height_y = 1;
    let width_y = yellow / height_y;
    let brown_calculated = calculateBrown(width_y, height_y);
    
    while (brown_calculated !== brown) {
        height_y++;
        
        while (yellow % height_y > 0) height_y++;
        
        width_y = yellow / height_y;
        brown_calculated = calculateBrown(width_y, height_y);
    }
    
    return [width_y + 2, height_y + 2];
    
    function calculateBrown(w, h) {
        return (w + 2) * 2 + h * 2;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
from math import sqrt

def solution(brown, yellow):
    for h in range(1, int(sqrt(yellow))+1):
        if yellow % h > 0:
            continue
        
        w = yellow // h
        
        if (w+2+h)*2 == brown:
            return [w+2, h+2]
        